src/main/java/org/broadinstitute/hellbender/tools/spark/BaseRecalibratorSpark.java

```diff
@@ -33,16 +33,68 @@
 import java.io.PrintStream;
 import java.util.List;
 
+/**
+ * Spark version of the first pass of the base quality score recalibration.
+ * Generates a recalibration table based on various covariates
+ * (such as read group, reported quality score, machine cycle, and nucleotide context).
+ *
+ * <p>
+ * This walker is designed to work as the first pass in a two-pass pipeline. It does a by-locus traversal operating
+ * only at sites that are not in dbSNP. We assume that all reference mismatches we see are therefore errors and indicative
+ * of poor base quality. This walker generates tables based on various user-specified covariates (such as read group,
+ * reported quality score, cycle, and context). Since there is a large amount of data one can then calculate an empirical
+ * probability of error given the particular covariates seen at this site, where p(error) = num mismatches / num observations.
+ * The output file is a table (of the several covariate values, num observations, num mismatches, empirical quality score).
+ * <p>
+ * Note: ReadGroupCovariate and QualityScoreCovariate are required covariates and will be added for the user regardless of whether or not they were specified.
+ *
+ * <p>
+ *
+ * <h3>Input</h3>
+ * <p>
+ * The input read data whose base quality scores need to be assessed.
+ * <p>
+ * A database of known polymorphic sites to skip over.
+ * </p>
+ *
+ * <h3>Output</h3>
+ * <p>
+ * A GATK Report file with many tables:
+ * <ol>
+ *     <li>The list of arguments</li>
+ *     <li>The quantized qualities table</li>
+ *     <li>The recalibration table by read group</li>
+ *     <li>The recalibration table by quality score</li>
+ *     <li>The recalibration table for all the optional covariates</li>
+ * </ol>
+ *
+ * The GATK Report is intended to be easy to read by humans or computers. Check out the documentation of the GATKReport to learn how to manipulate this table.
+ * </p>
+ *
+ * <h3>Examples</h3>
+ * <pre>
+ * ./gatk BaseRecalibratorSpark \
+ *   -I my_reads.bam \
+ *   -R resources/Homo_sapiens_assembly18.fasta \
+ *   --known-sites bundle/hg18/dbsnp_132.hg18.vcf \
+ *   --known-sites another/optional/setOfSitesToMask.vcf \
+ *   -o recal_data.table
+ * </pre>
```

Comment:
Again, I don't think this command will run as is. Have you tested these?

Reply:
Added spark arguments

Reply:
Thanks!